scipy/stats/_mstats_basic.py

```diff
@@ -336,14 +338,15 @@ def _mode1D(a):
         output = (ma.array(output[0]), ma.array(output[1]))
     else:
         output = ma.apply_along_axis(_mode1D, axis, a)
-        newshape = list(a.shape)
-        newshape[axis] = 1
-        slices = [slice(None)] * output.ndim
-        slices[axis] = 0
-        modes = output[tuple(slices)].reshape(newshape)
-        slices[axis] = 1
-        counts = output[tuple(slices)].reshape(newshape)
-        output = (modes, counts)
+        if keepdims is None or keepdims:
+            newshape = list(a.shape)
+            newshape[axis] = 1
+            slices = [slice(None)] * output.ndim
+            slices[axis] = 0
+            modes = output[tuple(slices)].reshape(newshape)
+            slices[axis] = 1
+            counts = output[tuple(slices)].reshape(newshape)
+            output = (modes, counts)
```

Comment:
This fix is still incomplete. With the example from the docstring of this function:
```
>>> mstats.mode(m_arr, keepdims=False)
ModeResult(mode=array([1.]), count=array([2.]))
```

Reply:
It looks like you added that example to the docstring in your last commit, which is why I hadn't tested it. `keepdims` was not added as a public parameter of `stats.mstats.mode`, only `stats.mode`. That would be a separate enhancement. I

I'll remove those new examples from the docstring and raise an error if any `kwargs` other than `_keepdims` is passed into `mstats.mode`. I would prefer not to add `keepdims` as a parameter of `stats.mstats.mode` as that would require other changes and tests.

Reply:
The way it was, the example code for `mstats.mode` is raising warnings though - that can't be right

Reply:
Yeah, I didn't notice that they were calling `stats.mode` in the docstring of `stats.mstats.mode`. No warnings are emitted by the doctests now.

Reply:
Ah okay, I missed the underscore in `kwargs.pop("_keepdims", True)`, I thought you intended for `keepdims` to be valid. I now see the `**kwargs` is only needed because `mstats.mode` is used from within `stats.mode`.

So this should be fine as is now. Let's leave the comment unresolved in case someone gets puzzled later and revisits this PR.

Reply:
Another way I could have done it is to move the implementation of `mstats.mode` into a private method called by both `mstats.mode` and `stats.mode`. that would keep `kwargs` out of the signature and avoid the need to manually produce the unexpected argument error. Want me to refactor?

Reply:
Sure, whatever you prefer. Let's leave that out of `1.9.x` indeed.